openlibrary/core/cache.py

```diff
@@ -191,7 +192,7 @@ def json_encode(self, value):
 
         memcache doesn't like spaces in the key.
         """
-        return json.dumps(value, separators=(",", ":"))
+        return json.dumps([] if isinstance(value, Nothing) else value, separators=(",", ":"))
```

Comment:
```suggestion
        return json.dumps([] if isinstance(value, Nothing) else value, separators=(",", ":"), cls=NothingEncoder)
```
`NothingEncoder` should handle any fields that have `Nothing` values.  It will have to be imported from `openlibrary.core.helpers `.